strength_percentiles.py

```diff
@@ -176,12 +199,189 @@ def populate_database(webpage, database):
                                              # individual's performance at the
                                              # meet
     for row in meet_results_table: # information for one lifter
-        result_and_categories = parse_result_and_categories(row)
+        result_and_categories = parse_row(row)
         if result_and_categories: # row contains results of meet
             cursor.execute(result_storage_string, result_and_categories)
 
     connection.commit()
     connection.close()
 
+####### Pandas functions
+def find_percentile(dataframe, lifts):
+    """
+    Finds the percentile of the input squat, bench, deadlift and total numbers
+    among the sample of results in the database.
+
+    Parameters:
+    -----------
+    dataframe: pandas dataframe
+        Dataframe containing squat, bench, deadlift, and/or total numbers
+        against which to compare the input numbers.
+    lifts: dictionary with string keys and float (or None) values
+        A dictionary that contains keys ('squat', 'bench', 'deadlift', 'total').
+        The values are floats representing someone's performance in the lift.
+
+    Returns:
+    --------
+    percentile: float
+        The user's percentile rank among the lifters in the database in the
+        input categories.
+
+    """
+    percentiles = []
+    for lift in (SQUAT, BENCH, DEADLIFT, TOTAL):
+        if lifts[lift]: # valid entry for lift
+            entered_lift = lifts[lift]
+            all_competitor_lifts = dataframe[lift]
+            number_of_lifts = all_competitor_lifts.count()
+            number_of_smaller_lifts = all_competitor_lifts[all_competitor_lifts < entered_lift].count()
+            percentile = (float(number_of_smaller_lifts)/number_of_lifts) * 100
```

Comment:
To do complete column-wise(or row-wise) operations, you generally want to use pandas library functions like http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.apply.html
You should be do something of the flavor:
percentiles = df.apply(lambda x: np.percentile(blah, ...), axis=...)